com.reprezen.swagedit/src/com/reprezen/swagedit/validation/SwaggerError.java

```diff
@@ -109,20 +128,66 @@ public String getMessage() {
 			builder.append("Failed to match exactly one schema:");
 			builder.append("\n");
 
-			for (String key: errors.keySet()) {
+			for (String location : orderedErrorLocations) {
 				builder.append(tabs);
 				builder.append(" - ");
-				builder.append(key);
+				builder.append(getHumanFriendlyText(location));
 				builder.append(":");
 				builder.append("\n");
 
-				for (SwaggerError e: errors.get(key)) {
+				for (SwaggerError e : errors.get(location)) {
 					builder.append(e.getMessage(true));
 				}
 			}
 
 			return builder.toString();
 		}
+
+		protected String getHumanFriendlyText(String location) {
+			JsonNode swaggerSchemaNode = findNode(location);
+			if (swaggerSchemaNode == null) {
+				return location;
+			}
+			JsonNode title = swaggerSchemaNode.get("title");
+			if (title != null) {
+				return title.asText();
+			}
+			JsonNode description = swaggerSchemaNode.get("title");
+			if (description != null) {
+				return description.asText();
+			}
+			// "$ref":"#/definitions/headerParameterSubSchema"
+			JsonNode ref = swaggerSchemaNode.get("$ref");
```

Comment:
We might have to be careful with Refs.  I remember seeing in a discussion that JSON Refs can use JSON Pointer as the locator, but they don't have to.  I think they may use a different form when it's an external reference.  Your algo may still work here, but it's a good idea to check on the JSON Ref specification.